src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -222,7 +376,10 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (!startsWithInheritDoc(ast)) {
+        if (containsSummaryTag(ast)) {
+            validateSummaryTag(ast);
+        }
```

Comment:
I did one more update:
and all UTs pass
```
✔ ~/java/github/romani/checkstyle [AkMo3/FixNewSummaryFormat L|✚ 2] 
05:19 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
index 808749d..29310e3 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
@@ -382,13 +382,13 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
                 log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
             }
             else if (!period.isEmpty()) {
-                final String firstSentence = getFirstSentence(ast);
-                final int endOfSentence = firstSentence.lastIndexOf(period);
+                //final String firstSentence = getFirstSentence(ast);
+                final int endOfSentence = summaryDoc.lastIndexOf(period);
                 if (!summaryDoc.contains(period)) {
                     log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
                 }
                 if (endOfSentence != -1
-                        && containsForbiddenFragment(firstSentence.substring(0, endOfSentence))) {
+                        && containsForbiddenFragment(summaryDoc.substring(0, endOfSentence))) {
                     log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
                 }
             }
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
index c344c8b..7fa31e3 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
@@ -78,7 +78,6 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "53: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "58: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "69: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "83: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
             "103: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "116: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "121: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
```

and what I driving to is validation of summary should be identical for old style and special tag.
the only difference is how you get summary content, so difference should be only in 
`final String summaryDoc = getSummarySentence(ast);` line.

Reply:
> Validation of summary should be identical for old-style and special tag.

Are we then not concerned about inline tags like {@throws} that are not allowed inside a Javadoc? Old style already doesn't detect Javadoc inside HTML tags, see https://github.com/checkstyle/checkstyle/issues/9806, neither in conforms with https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html where it is specifically mentioned which tags can be used inside overview.

Reply:
> Are we then not concerned about inline tags like {@throws} that are not allowed inside a Javadoc?

yes, This Check should not care about this. If we start to care about this , it should be separate Check on this, but we should rely on javadoc tool to validate this. Checkstyle is not substitution of javadoc tool validations, it does validation in addition.

We even prone to remove Checks that do validation that modern jdk treat as compilation error.

> as we are covering almost all possible cases here.

sorry I have doubt on this. The only reason we have so much bug in Checkstyle is because user always find a way to write code is such a way that our validation do false-positive or false-negative.

-----

please remove this validation from Check. Sorry about this.

Reply:
please apply my path https://github.com/checkstyle/checkstyle/pull/9174#discussion_r612396057 to your PR.

Reply:
Sir, I tried to implement your approach but it fails on input test cases like:
```
  /**
   * {@summary first sentence is normally the summary.
   * Use of html tags:
   * {@code SomeCodeHere.}
   * <ul>
   * <li>Item one.</li>
   * <li>No period here</li>
   * </ul>}
   */
  private void invalidInlineJavadocList() // violation
  {
  }
```
As `final int endOfSentence = summaryDoc.lastIndexOf(period);` would return the period of first `.` rather then checking the end. So, I am keeping my method of checking the end. Rest of the validation of summary is same.

Reply:
Yes, we need more smart validation of last period position. It should be last html printable symbol.

Reply:
This is being done by `isPeriodAtEnd()`, where I am checking if last element is `period` of `getVisibleContent`.

Reply:
this become a bit more complicated i moved this request to https://github.com/checkstyle/checkstyle/issues/9912 , it will be done separately. lets focus in this PR on summary tag.